### DISCUSSION TITLE
Set proxy for a single request

### DISCUSSION THREAD

**Discussion by AneryCoft** [encode/httpx#3183] [General]
We can only create a global proxy when the client is instantiated.
Whether to add proxy parameters in every client requests？

**Comment by lovelydinosaur** [comment]
So, `httpx` does not currently allow setting proxies per-request. You need to configure a client specifically for the request.

If anyone is interested in understanding *why* this is the case I'd suggest taking a look at [the `Transport` API](https://www.python-httpx.org/advanced/transports/#custom-transports), and [implementation of the default HTTP transport](https://github.com/encode/httpx/blob/master/httpx/_transports/default.py).

Proxy configuration is dealt with *on instantiating the transport*, since the `handle_request(Request) -> Response` API is kept deliberately simple.

I'm happy to discuss more with anyone who's interested how we might go about resolving this, if anyone is keen on an in-depth design discussion. *(fun)* 🥳
